src/app/settings/store/settings.effects.ts

```diff
@@ -0,0 +1,77 @@
+import { Injectable } from '@angular/core';
+import { Actions, Effect, ofType } from '@ngrx/effects';
+import { Action, Store } from '@ngrx/store';
+import { Observable } from 'rxjs/Observable';
+import { map } from 'rxjs/operators/map';
+import { first } from 'rxjs/operators/first';
+import { mergeMap } from 'rxjs/operators/mergeMap';
+
+import { State, UserTagsActions } from '../../root-store';
+import { SettingsActionTypes, UpdateSettings } from './settings.action';
+import { SettingsViewModel } from '../view-models';
+import { getSettingsViewModel } from './settings.selectors';
+
+interface Change {
+  key: string,
+  value: any
+}
+
+function getDifferentKeysAndValues(current: SettingsViewModel, changed: SettingsViewModel): Change[] {
```

Comment:
think about using some lodash method

Reply:
Can't find any lodash method that will simplify this part of the code. Instead, I added tests for this function.